.github/workflows/doc_check_on_pr.yml

```diff
@@ -50,9 +50,11 @@ jobs:
       - name: Make the merged branch main
         run: |
           cd ColossalAI
-          curBranch=$(git rev-parse --abbrev-ref HEAD)
-          git checkout main
-          git merge $curBranch             # fast-forward master up to the merge
+          git checkout -b main
+          git branch -u origin/main
+          git config user.name 'github-actions'
+          git config user.email 'github-actions@github.com'
```

Comment:
I think the reason why the current merged branch needs to be set to to main is that the documentation will treat the main branch as the `current` version. The whole workflow is like this:

1. the github actions `checkout` will merge the source branch and the target branch, this branch can be obtained from `$(git rev-parse --abbrev-ref HEAD)`, let's name this branch as `curBranch`
2. make this `curBranch` the main branch
3. documentation build will then take the main branch to build documentation

Not sure whether anything wrong with the merge?
5.

Reply:
As https://github.com/hpcaitech/ColossalAI/actions/runs/4977382383/jobs/8906746190?pr=3729#step:5:14, it shows `Already up to date.` when merging. So no change is applied.

Reply:
We can directly use the merged branch (`curBranch`) as the main branch.

Reply:
Nice!